### PR TITLE
Added a custom hook via GMaps to `Preview` the location!

### LINKED ISSUE #10: Add `Preview` of the `lat:long` via GMaps
# Current State : 

**_No method to preview the gathered `lat:long`_**

* +1 : **User is Blind**, if the data was correct or not.
* +2 : **User is Blind**, how precise the location was gathered.

# Expected State : 

**There should be !!**

## Are you assigned to this issue ?

**_Heck Ya!!_** :sunglasses:

### PR DESCRIPTION
**Pull request by 0x48piraj**
# Fixes #10 

## Changes Done : 

* Added **custom hook** to GMaps :fire: _(bypassed the messy `api` method, with my own workaround.)_ [**~5hrs of researching**]
* Fixed names, Fixed variables
* Enhanced hook, added a hyperlink.
* Fixed `<br>` bug, that caused improper rendering of locations. :100: 
* Added `target` tag to open in separate page, to avoid reloading, going back, or any possible headache.

**_Test-Link : https://0x48piraj.github.io/geolocation/_**

## Screenshot : 

image

### CODE REVIEW

**Comment by 0x48piraj** [comment]
Ping @akshatnitd 

> The Travis is failing not because of my change!:smile:

image

**Comment by 0x48piraj** [comment]
@abishekvashok 
I intentionally did that, because if we'll open the GMaps on the very same page, we'll have to reload the page and will have to do all things again!

**Comment by abishekvashok** [comment]
We don't need to reload to place a marker in a map. Make the map hidden.

**Comment by 0x48piraj** [comment]
Pardon me, but I didn't got what you meant @abishekvashok 
Can you explain ?

**Comment by abishekvashok** [comment]
Make a short map which is visible after the user clicks the button like what is there in the gci17.fossasia.org website. if possible please use open street maps to minimize use of priority software and foster open source.

**Comment by 0x48piraj** [comment]
I'll try to implement that @abishekvashok :+1:

**Comment by 0x48piraj** [comment]
@akshatnitd @abishekvashok 
**Thank you very much !!**

I integrated **OSM**! Thanks again @abishekvashok :100: 
It's now looking so good :heart_eyes_cat:

**Comment by 0x48piraj** [comment]
**_Test-Link : https://0x48piraj.github.io/geolocation/_**
image
## Test it out !

File: index.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
     <meta http-equiv="X-UA-Compatible" content="chrome=1">
     <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
     <link href="css/stylesheet.css" rel="stylesheet">
+	<link href="https://unpkg.com/leaflet@1.0.1/dist/leaflet.css" rel="stylesheet"/>
```
Comment (akshatnitd) [inline index.html]: Indent

File: index.html [hunk 2]
```diff
@@ -26,5 +29,6 @@ <h2>Geo locator to faciliate easy integration of location data</h2>
     </div>
     <div class="card" id="info">Created with ♡ for FOSSASIA</div>
     <script src="js/main.js"></script>
+	<script src="https://unpkg.com/leaflet@1.0.1/dist/leaflet.js"></script>
```
Comment (akshatnitd) [inline index.html]: Indent

File: index.html (hunk 1 above)
Comment (akshatnitd) [inline index.html]: Indent

File: index.html (hunk 2 above)
Comment (akshatnitd) [inline index.html]: Indent

File: js/main.js [hunk 3]
```diff
@@ -14,6 +14,17 @@ function rit(position) {
     var mylocation = "lat: " + position.coords.latitude +"<br>";
     mylocation += "lng: " + position.coords.longitude+"<br><br>";
     output.innerHTML = mylocation;
+    my_preview = "https://www.google.com/maps/?q=" + position.coords.latitude + "," +position.coords.longitude;
+    preview.innerHTML = "<a target='_blank' href='"+my_preview+"'>PREVIEW on GMaps!</a>"+"<br><br>";
+	var element = document.getElementById('osm-map');
```
Comment (akshatnitd) [inline js/main.js]: Indent

File: js/main.js [hunk 4]
```diff
@@ -14,6 +14,17 @@ function rit(position) {
...
+	element.style = 'height:300px;';
+	var map = L.map(element);
+	L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png', {
+        attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
```
Comment (vaibhavsingh97) [inline js/main.js]: Any reason for using OpenStreetMap? You can use Google Maps instead

File: js/main.js [hunk 5]
```diff
@@ -14,6 +14,17 @@ function rit(position) {
     var mylocation = "lat: " + position.coords.latitude +"<br>";
     mylocation += "lng: " + position.coords.longitude+"<br><br>";
     output.innerHTML = mylocation;
+    my_preview = "https://www.google.com/maps/?q=" + position.coords.latitude + "," +position.coords.longitude;
+    preview.innerHTML = "<a target='_blank' href='"+my_preview+"'>PREVIEW on GMaps!</a>"+"<br><br>";
```
Comment (vaibhavsingh97) [inline js/main.js]: Why we need extra link? User can click on the marker and we can rediret to the Google Maps

File: js/main.js (hunk 4 above)
Comment (abishekvashok) [reply js/main.js]: @vaibhavsingh97 it's open source, we need to move away from priority software and encourage open source

**Comment by 0x48piraj** [comment]
Closing stale PR, re-opening a PR which takes care of this PR and the related issue.

**Pull request closed without merging**